src/web/web.ts

```diff
@@ -1,18 +1,25 @@
 import express from "express";
-import http from "http";
+import http, {Server} from "http";
 import path from "path";
 import RoomController from "./backend/controllers/room.controller";
 import Controller from './backend/interfaces/controller.interface';
 import AuthenticationController from "./backend/authentication/authentication.controller";
 import bodyParser from "body-parser";
 import { UserController } from "./backend/controllers/user.controller";
 import cors from "cors";
+import io from "socket.io"
+import SocketIO from "socket.io";
 
 class App {
   public app: express.Application;
+  private server: Server;
+  private socket: SocketIO.Server;
   // Express app initialization
   constructor() {
     this.app = express();
+    this.server = http.createServer(this.app);
+    this.server.listen(3001);
```

Comment:
we have initializatiin logic in listen method in same file, shouldn't override it

Reply:
fixed